docs/user_guide.rst

```diff
@@ -1,2 +1,28 @@
 User Guide
 ==========
+
+.. contents:: :depth: 3
+
```

Comment:
I feel like we need at least one or two sentences of introduction before jumping in with an example. Perhaps repeat what FoxPuppet is for.